include/project_parser.hpp

```diff
@@ -209,6 +209,7 @@ struct Project {
     const Project *root() const;
     bool cmake_minimum_version(int major, int minor) const;
     static bool is_condition_name(const std::string &name);
+    bool is_valid_hex(std::string value);
```

Comment:
Not important (see other comment), but as a general rule you should pass strings by const reference unless there is a good reason not to. This will make unnecessary copies when calling the function.